Server/routes/comments.js

```diff
@@ -0,0 +1,249 @@
+const express = require('express');
+const router = express.Router();
+
+const {
+    getAllCommentsByPostId,
+    addCommentToPost,
+    editComment,
+    deleteComment
+} = require('../queries/comments');
+
+const { authenticateUser } = require('../queries/authentication')
+
+const handleError = (response, err) => {
+    if (err.message === "No data returned from the query.") {
+        response.status(404)
+        response.json({
+            status: 'fail',
+            message: 'Unexpected route',
+            payload: null,
+        })
+    } else if (err.constraint === 'comments_post_id_fkey') {
+        response.status(404)
+        response.json({
+            status: 'fail',
+            message: 'Wrong route',
+            payload: null,
+        })
+    } else { 
+        console.log(err)
+        response.status(500)
+        response.json({
+            status: 'fail',
+            message: 'Sorry, Something Went Wrong (BE)',
+            payload: null,
+        })
+    }
+}
+
+const isValidId = (id) => {
+    if (!isNaN(parseInt(id)) && id.length === (parseInt(id) + '').length) {
+        return true
+    }
+    return false
+}
+
+const isUserAllowed = async (response, id, password) => {
+    try {
+        return await authenticateUser(id, password)
+    } catch (err) {
+        console.log(err)
+        if (err.message === "No data returned from the query.") {
+            return false
+        } else {
+            handleError(response, err)
+        }
+    }
+}
+
+// GET ALL COMMENTS BY POST ID
+router.get('/:postId', async (request, response) => {
+    const postId = request.params.postId;
+    const validId = isValidId(postId);
+
+    if (!validId) {
+        response.status(404)
+            response.json({
+                status: 'fail',     
+                message: 'Wrong route',
+                payload: null,
+            })
+    } else {
+        try {
+            const allCommentsByPostId = await getAllCommentsByPostId(postId);
+            if (allCommentsByPostId.length) {
+                response.json({
+                    status: 'success',
+                    message: `Successfully retrieved all comments related to the post: ${postId}`,
+                    payload: allCommentsByPostId,
+                })
+            } else {
```

Comment:
what if the post exists but there are just no comments? i dont think the 400 bad request is the best response because it will add complexity dealing with on the front end

Reply:
would this be a 400 or 404?

Reply:
GOod catch